spec/proxy_spec.lua

```diff
@@ -80,4 +80,36 @@ describe('Proxy', function()
       assert.same(8080, get_upstream({ api_backend = 'http://example.com:8080' }).port)
     end)
   end)
+
+  describe('.authorize', function()
+    local authorize = proxy.authorize
+    local service = { backend_authentication = { value = 'not_baz' } }
+    local usage = 'foo'
+    local credentials = 'client_id=blah'
+
+    it('takes ttl value if sent', function()
+      local ttl = 80
+      ngx.var = { cached_key = credentials, usage=usage, credentials=credentials, http_x_3scale_debug='baz', real_url='blah' }
+      ngx.ctx = { backend_upstream = ''}
+      ngx.shared.api_keys = { cached_key = '', get = function () return {} end }
```

Comment:
I think here it should do the same as for `ngx.var`.

```lua
ngx.shared = { api_keys = ... }
```

Because otherwise you are mutating the `shared` table and leaking it to another tests .

Also I don't see why there is `cached_key = ''`